armada-ths/ais
# Api questions

Implemented /api/questions api.
The formats are described in https://docs.google.com/document/d/1nbyse_gzmiNQT4MaSW5de2vwaqWDXCICs5J3fC4wafw document, I also tested the api to be able to handle illegal put payloads without crushing (if you manage to get a status code different from 200, 400 or 406 - tell me, as it's not intended!)

to test:
* run tests
* make sure you have a survey linked to current Fair
* create at least one student question
* create at least one work field area
* create at least one work field
* make sure 3 of the above are linked to the survey
* goto localhost:8000/api/questions
* try to make a correct request (similar to the one described in the provided document) and if the request is correct you should get a "Answers submitted!" response!
* check that the answers were saved correctly in admin view

Note 1: you will need to send PUT request relative to specific student (api/quiestion?student_id=SOME_NUMBER), if a student_profile with that ID doesn't exist a new one will be created.
Note 2: it's possible to send only answers/only fields to api/questions and it would still be considered a valid request.
Note 3: it's also legal to send only answers to some of the questions, but the areas should always be all selected ones!
